### DISCUSSION TITLE
Servicestack Server Sent Events

### DISCUSSION THREAD

**Discussion by Pinox** [ServiceStack/Discuss#109] [Q&A] [answered]
Hi @mythz 

I'm currently using SS server sent events.  With all the action from SS where asp.net identity is now supported within SS I'm wondering if there are any plans in the future to move to signalr also?

Basically SignalR SSE is now very similar to SS SSE especially now that asp.net identity auth are also used in SS.

var hubConnection = new HubConnectionBuilder()
            .WithUrl( Navigator.ToAbsoluteUri( hubPath ), options =>
            {
                // using securitytoken
                options.AccessTokenProvider = () => Task.FromResult( ((MyAuthStateProvider)AuthStateProv).Token );
                options.Transports = Microsoft.AspNetCore.Http.Connections.HttpTransportType.ServerSentEvents;
                options.SkipNegotiation = true;
            })
            .Build();onnection.KeepAliveInterval = TimeSpan.FromSeconds( 15 ); 

What's your current view regarding SignalR and SS SSE regarding future integration with products like Aspire.

Thanks.

**Comment by mythz** [comment, marked as answer]
The primary reason behind switching to Identity Auth, ASP .NET Core IOC, Endpoint Routing, System.Text.Json, etc. is so it integrates fully with ASP .NET Core IOC Apps so you can mix and choose whatever technology you like without conflicts.

There's nothing to "Move to" with SignalR, it's a different client and server implementation operating at a different endpoint, you should be able to use it the same way it's used everywhere else. We don't intend to develop our own client for it, you'd continue using Microsoft's recommended SignalR client for the platform you want to use it from.

We've started looking at Aspire, IMO it's only useful for developing distributed App's which none of our templates really target so we're not sure what deliverables we'll have for it yet, whether it's just a Demo App or Blog Post or Video, etc.

**Reply by Pinox** [reply]
Thanks @myth.  By "move to" I mean recommended path by SS.

I have not tried Aspire, waiting for it to be stable ;)) I do like what seems like an easy integration into Graphana that produces nice graphs for useful metrics on endpoints / Sql server request , redis , health checks ,error rates etc.  Hopefully in time we can also use it for integrating into cloud service providers.

![image](https://github.com/ServiceStack/Discuss/assets/5297255/b01af0bb-5f49-40b6-813c-1eca73d09325)

**Reply by mythz** [reply]
Not having used SignalR in any real-world capacity I can't recommend it personally, but I don't not recommend it if it suitable for your use-case. The great thing is that you should be able to use which ever parts you like without conflict.

**Reply by Pinox** [reply]
Yea my use case is unchanged, as they say "If it ain't broke, don't fix it".
